lib/validators.rb

```diff
@@ -137,9 +137,107 @@ def check_shipments_specificities
       }
     end
 
+    def vehicle_days_availability(vehicles_set, timewindows_set)
+      vehicles_days = timewindows_set.collect{ |timewindow|
+        if timewindow.empty?
+          []
+        else
+          days = timewindow.flat_map{ |tw| tw[:day_index] }
+          days.compact!
+          days.uniq
+        end
+      }.delete_if(&:empty?)
+
+      vehicles_unavailable_indices = vehicles_set.collect{ |v| v[:unavailable_work_day_indices] }
+      vehicles_unavailable_indices.compact!
+
+      [vehicles_days, vehicles_unavailable_indices]
+    end
+
+    def check_vehicle_trips_stores_consistency(relation_vehicles)
+      relation_vehicles.each_with_index{ |vehicle_trip, v_i|
+        case v_i
+        when 0
+          unless vehicle_trip[:end_point_id]
+            raise OptimizerWrapper::DiscordantProblemError.new('First trip should at least have end point id')
+          end
+        when relation_vehicles.size - 1
+          unless vehicle_trip[:start_point_id]
+            raise OptimizerWrapper::DiscordantProblemError.new('Last trip should at least have start point id')
+          end
+        else
+          unless vehicle_trip[:start_point_id] && vehicle_trip[:end_point_id]
+            raise OptimizerWrapper::DiscordantProblemError.new('Trip should have start and end point id')
+          end
+        end
+
+        next if v_i.zero? ||
+                vehicle_trip[:start_point_id] == relation_vehicles[v_i - 1][:end_point_id] ||
+                (@hash[:points].find{ |pt| pt[:id] == vehicle_trip[:start_point_id] }[:location] ==
+                  @hash[:points].find{ |pt| pt[:id] == vehicle_trip[:end_point_id] }[:location])
```

Comment:
this looks like a bug :thinking:  shouldn't it be `relation_vehicles[v_i - 1][:end_point_id]` instead of `vehicle_trip[:end_point_id]` ?

Reply:
Right ! Thank you !

Reply:
Do you think _find_ is costly enough to keep this line ? 
```
                  vehicle_trip[:start_point_id] == relation_vehicles[v_i - 1][:end_point_id] ||
```
because in fact this does exactly the same.

Reply:
I thought the exact same thing :D haa 

I think your way is good since it does the least costly check first and the other one only if it is necessary.

Since it is a find in all points, it might be depending on the size of the problem (and since this is all inside the POST process, anything faster (as long as it is correct) is better)